airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalAttemptExecution.java

```diff
@@ -165,6 +168,25 @@ public OUTPUT get() {
     }
   }
 
+  private void retryWithJitter(final AirbyteApiClient airbyteApiClient) throws InterruptedException {
+    final int maxRetries = 3;
+    int currRetries = 0;
+    boolean needToSend = true;
+
+    while (needToSend && currRetries < maxRetries) {
+      try {
+        LOGGER.info("Attempt {} to save workflow id", currRetries);
+        saveWorkflowIdForCancellation(airbyteApiClient);
```

Comment:
@pmossman @xiaohansong very naive implementation - this is tough to reproduce so I'm hoping to test this in Prod to see if it's effective. 

Worse come to worse, this isn't and we are right back where we started.

Do let me know if people aren't comfortable with this quick fix.

Reply:
I think this looks like a good way to test in prod. I would maybe consider adding a fourth and final retry that waits something like 5 minutes instead of 10 seconds (and maybe log whenever we hit that case). This way, if there's some sort of network transiency that lasts for longer than 30 seconds, but resolves itself within 5 minutes, we have some extra info about the class of problem we might be facing.

Reply:
done - testing this now